src/core/server/config/schema/types/string_type.ts

```diff
@@ -24,6 +24,7 @@ import { Type, TypeOptions } from './type';
 export type StringOptions = TypeOptions<string> & {
   minLength?: number;
   maxLength?: number;
+  hostname?: boolean;
```

Comment:
__note:__ another option would be introducing separate `hostname` type if we want, I'm just not sure whether we need this, I see both pros and cons. What would you prefer?

Reply:
I don't have strong opinions one way or another. The big downside here is the slipper slope effect, where StringType starts validating things like a bunch of stuff like emails, ip addresses, etc. But if/when we start going down that route, we can always refactor. It's probably worth not over-architecting right now.

Reply:
> It's probably worth not over-architecting right now.

Yeah, that was my thought too - it's pretty easy to change if we decide to do so.

Reply:
Yeah, I was thinking that at some point I'd love if we supported a `path` type, one that would resolve relative paths based on the location of the config file containing the value, but if we added that to string then we'd have to validate that people didn't include both `hostname: true` and `path: true`.

Reply:
> but if we added that to string then we'd have to validate that people didn't include both `hostname: true` and `path: true`.

Not necessarily, we can just change `{ hostname: boolean }` to something like `{ special: StringSpecial.Hostname | StringSpecial.Path | StringSpecial.IPAddress }` and use it like this:
```js
schema.string({ special: StringSpecial.Hostname })`;
```
If we need some additional args, we can use object that follows some interface instead of enum. But with args we can start thinking about dedicated type. A plenty of options with our thin wrapper around Joi :)